File: generatediary.pl
#!/usr/bin/perl -w

my $status = `git log`;
my $regex = qr/"Date.*"/;
# print $status;
# open $line, "<",\$status;

# while ($var = <$line>)
# {
# 	chomp $var;
# 	print $var,"\n";
# 	$var =~ s/"Sep"/"XXXXXXX"/g;
	
# 	# if ($var =~ /".*Date.*"/)
# 	# {print $1;}
# 	# else {}
# 	# {print "$var";}
# }

@statarray = split /"\n"/, $status;
# print @statarray;
foreach	$tmp (@statarray)
{
	if ($tmp =~ /$regex/)
	{print $tmp;}
	else {}
	# {print "$tmp";}

}

File: generatediary.py
#!/usr/bin/python

import subprocess, sys, re, string

# Courtesy of Stack Overflow 
# git log --pretty=format:"%h%x09%an%x09%ad%x09%s"

gitcmd = "git log --pretty=format:\"%x01%an%x09%ad%x09%s\"";

# Courtesy of Stack Overflow 
# http://stackoverflow.com/questions/4256107/running-bash-commands-in-python

log = subprocess.Popen(gitcmd.split(), stdout=subprocess.PIPE)
output = log.communicate()[0]
regex = ".*Date(.*)"
diary = open ('diary.txt','w')

for line in output.splitlines():
	# if re.match(regex,line):
	line = re.sub('"','',line)
	line = re.sub('^\ +','',line)
	print line
	diary.write(line+"\n")